src/tests/ftest/pool/create_all_hw.py

```diff
@@ -0,0 +1,319 @@
+        self.pool[0].get_info()
+        tier_bytes = [self.pool[0].info.pi_space.ps_space.s_total]
+        self.assertLessEqual(abs(self.scm_avail_bytes - 2 * tier_bytes[0][0]), self.delta_bytes,
+                "Invalid SCM size: want={}, got={}, delta={}".format(self.scm_avail_bytes / 2,
+                    tier_bytes[0][0], self.delta_bytes))
+        self.assertLessEqual(abs(self.smd_avail_bytes - 2 * tier_bytes[0][1]), self.delta_bytes,
+                "Invalid SMD size: want={}, got={}, delta={}".format(self.smd_avail_bytes / 2,
+                    tier_bytes[0][1], self.delta_bytes))
+
+        self.log.info("Checking the distribution of the first pool")
+        self.check_pool_distribution()
+
+        self.scm_avail_bytes, self.smd_avail_bytes = self.get_pool_available_bytes()
+
+        self.log.info("Creating a second pool with 100% of the available storage")
+        self.pool[1].create()
+        self.pool[1].get_info()
+        self.assertEqual(self.pool[1].dmg.result.exit_status, 0,
+                "Second pool could not be created")
+
+        self.pool[1].get_info()
+        tier_bytes.append(self.pool[1].info.pi_space.ps_space.s_total)
+
+        self.log.info("Checking size of the second pool with the old available size")
+        self.assertLessEqual(abs(self.scm_avail_bytes - tier_bytes[1][0]), self.delta_bytes,
+                "Invalid SCM size: want={}, got={}, delta={}".format(self.scm_avail_bytes,
+                    tier_bytes[1][0], self.delta_bytes))
+        self.assertLessEqual(abs(self.smd_avail_bytes - tier_bytes[1][1]), self.delta_bytes,
+                "Invalid SMD size: want={}, got={}, delta={}".format(self.smd_avail_bytes,
+                    tier_bytes[1][1], self.delta_bytes))
+
+        self.log.info("Checking size of the second pool with the size of the first pool")
+        scm_delta_bytes = self.ranks_size * self.max_scm_metadata_bytes + self.delta_bytes
+        self.assertLessEqual(abs(tier_bytes[0][0] - tier_bytes[1][0]), scm_delta_bytes,
+                "Invalid SCM size: want={}, got={}, delta={}".format(tier_bytes[0][0],
+                    tier_bytes[1][0], self.delta_bytes))
+        smd_delta_bytes = self.smd_device_size
+        smd_delta_bytes *= self.engine_target_size - 1
+        smd_delta_bytes *= self.smd_cluster_bytes
+        smd_delta_bytes += self.delta_bytes
+        self.assertLessEqual(abs(tier_bytes[0][1] - tier_bytes[1][1]), smd_delta_bytes,
+                "Invalid SMD size: want={}, got={}, delta={}".format(tier_bytes[0][1],
+                    tier_bytes[1][1], self.delta_bytes))
+
+        self.scm_avail_bytes, self.smd_avail_bytes = self.get_pool_available_bytes()
+
+        self.log.info("Checking size of available storage after the creation of the second pool")
+        self.assertLessEqual(self.scm_avail_bytes, self.delta_bytes,
+                "Invalid SCM size: want=0, got={}".format(self.scm_avail_bytes))
```

Comment:
This is minor. I think what we want is not 0, but less than `self.delta_bytes`.

Reply:
- [ ] Fix invalid assert message